### PR TITLE
Refactoring: cleanup

### PR DESCRIPTION
**Pull request by toaster**
### Description:

I found some focus artifacts in `PopUpMenu`.
I removed them and it turned out that `testCanvas` had no proper focus support.
I then moved the overlay focus manager handling into the `internal.OverlayStack`. Thus, the test canvas can make use of it. In addition I copied the simple focus manager logic from `glCanvas` into `testCanvas` which is sufficient to make the tests happy again.

### Checklist:

- [ ] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

### CODE REVIEW

File: test/testcanvas.go [hunk 1]
```diff
@@ -43,90 +44,109 @@ type testCanvas struct {
...
 
-func (c *testCanvas) SetContent(content fyne.CanvasObject) {
-	c.propertyLock.Lock()
-	c.content = content
-	c.propertyLock.Unlock()
-
-	if content == nil {
-		return
+// LaidOutObjects returns all fyne.CanvasObject starting at the given fyne.CanvasObject which is laid out previously.
+func LaidOutObjects(o fyne.CanvasObject) (objects []fyne.CanvasObject) {
```
Comment (andydotxyz) [inline test/testcanvas.go:57]: Seems like a strange place to move this function to

**Review by andydotxyz — COMMENTED** [review]
Looks good, though the rearranged function order seems strange

File: test/testcanvas.go (hunk 1 above)
Comment (toaster) [reply test/testcanvas.go:57]: I thought we agreed to exported package functions before any package type functions, not?

Comment (andydotxyz) [reply test/testcanvas.go:57]: Pretty sure it’s type -> constructor -> exported type func -> unexported type func.
We didn’t put exported package funcs in the list I don’t think, I would imagine that it’s either at the top or the bottom of the file, but not in between type declarations and type functions.

Comment (toaster) [reply test/testcanvas.go:57]: Okay, so I propose the top of the file :).

**Review by stuartmscott — APPROVED** [review]

**Pull request merged**